ccw.core/src/java/ccw/CCWPlugin.java

```diff
@@ -234,6 +234,17 @@ private final void logDependenciesInformation(final BundleContext context) {
 
     public CCWPlugin() { }
 
+    private volatile int bundleState = Bundle.UNINSTALLED;
+
+    /**
+     * Return if the bundle is going to stop.
+     * @return True or false.
+     */
+    public static boolean isBundleStopping() {
+        CCWPlugin plugin = CCWPlugin.getDefault();
+        return plugin != null ? plugin.bundleState == Bundle.STOPPING : false;
```

Comment:
Yes probably I should change this method to some ```canLoadWithBundle``` because you are right, there are many more cases where we don't want to load...don't know why I didn't think about it ...